zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeperMain.java

```diff
@@ -289,6 +302,17 @@ protected void connectToZK(String newHost) throws InterruptedException, IOExcept
             System.setProperty(ZKClientConfig.SECURE_CLIENT, "true");
             System.out.println("Secure connection is enabled");
         }
+        if (cl.getOption("waitforconnection") != null) {
+            connectLatch = new CountDownLatch(1);
+        }
+        int timeout = Integer.parseInt(cl.getOption("timeout"));
+        zk = new ZooKeeperAdmin(host, timeout, new MyWatcher(),readOnly);
+        if (connectLatch != null) {
```

Comment:
Could you please add more clarifications about the difference between `waitforconnection` and `timeout` in the GitHub description?

Reply:
I have given some clarification on waitforconnection and timeout options in GitHub description. Thanks